engine/cluster/src/test/java/org/opennms/oce/engine/cluster/ClusterEngineTest.java

```diff
@@ -313,55 +317,75 @@ public void canHandleAlarmsInClusters() {
                 .setId(Long.valueOf(situationIdGenerator.incrementAndGet()).toString())
                 .addAlarm(alarm1InSpaceTime.getAlarm())
                 .build();
-        Map<String, Situation> alarmIdToSituationMap = new ImmutableMap.Builder<String, Situation>().put("1", existingSituation)
-                .build();
-        Map<String, Situation> situationsById = new ImmutableMap.Builder<String, Situation>()
-            .put(existingSituation.getId(), existingSituation)
-                .build();
-
-        situations = engine.mapClusterToSituations(cluster, alarmIdToSituationMap, situationsById, 0L);
+        engine.setSituations(Collections.singleton(existingSituation));
+        context = engine.getTickContextFor(0L);
+        engine.mapClusterToSituations(cluster, context);
+        situations = context.getNewOrUpdatedSituations();
         assertThat(situations, hasSize(1));
-        assertThat(Iterables.getFirst(situations, null), sameInstance(existingSituation));
-        assertThat(Iterables.getFirst(situations, null).getAlarms(), hasSize(2));
+        Situation updatedSituation = Iterables.getFirst(situations, null);
+        assertThat(updatedSituation.getAlarms(), hasSize(2));
 
         // A cluster with alarms that are already mapped to separate situations should not updated/create any situations
         Situation existingSituation2 = ImmutableSituation.newBuilderNow()
                 .setId(Long.valueOf(situationIdGenerator.incrementAndGet()).toString())
                 .addAlarm(alarm2InSpaceTime.getAlarm())
                 .build();
-        alarmIdToSituationMap = new ImmutableMap.Builder<String, Situation>().put("1", existingSituation).put("2", existingSituation2)
-                .build();
-        situationsById = new ImmutableMap.Builder<String, Situation>().put(existingSituation.getId(), existingSituation)
-            .put(existingSituation2.getId(), existingSituation)
-                .build();
-
-        situations = engine.mapClusterToSituations(cluster, alarmIdToSituationMap, situationsById, 0L);
+        engine.setSituations(Arrays.asList(existingSituation, existingSituation2));
+        context = engine.getTickContextFor(0L);
+        engine.mapClusterToSituations(cluster, context);
+        situations = context.getNewOrUpdatedSituations();
         assertThat(situations, hasSize(0));
+    }
 
-        // If a cluster contains alarms in different situations and one or more alarms that are not
-        // yet associated to a situation, then we chose to add the alarms without situations to their the same situation
-        // as their closest neighbor
-        existingSituation = ImmutableSituation.newBuilderNow()
+    @Test
+    public void canUpdateExistingSituationsWithManyAlarmsInCluster() {
+        // Create 4 unique alarms
+        final BiFunction<Integer,Integer,AlarmInSpaceTime> createAlarmWithIdOnNodeId = (alarmId, nodeId) -> {
```

Comment:
Just my opinion but I think its more readable for lambdas like this one to become real methods.